tool/microkit/src/lib.rs

```diff
@@ -123,15 +123,12 @@ impl MemoryRegion {
     }
 }
 
+#[derive(Default)]
```

Comment:
Default doesn't really make sense for DisjointMemoryRegion. Can you keep this as `new`?

Reply:
I can do so. I do not have a credible source, but I got the impression that having `::new()` that does nothing but initialize the struct with default values for its members is an anti-pattern. I would only use `new()` if it takes arguments and does some sort of non-default initialization. Or to set non-public members to a non-default value. If it does not take arguments but does non-default initialization, and has some side-effects, `::new()` is also fine. But if `::new()` really does not do anything, no side-effects, no arguments, no non-default initialization, where is the benefit in not saving a couple of lines and just using `[derive(Default)]`? That has also the benefit that it unlocks some syntax constructs like the `MyStruct { some_member: 4, some_other_attr: 1, .. Default::default()}` to initialize most members with default values but a few.

TL;DR, I don't agree with your statement, and from a technical perspective think its less idiomatic. I can not provide strong evidence for my point. I do not want to argue for long, so if you insist I will comply without further arguments :smile: .

Reply:
Ok sure. I really don't care that much so let's just leave it with `Default`.